record.go

```diff
@@ -193,17 +193,20 @@ func ParseRecord(str string, w http.ResponseWriter, req *http.Request, c Config)
 		r.Code = http.StatusFound
 	}
 
-	if r.Type == "" {
-		r.Type = "host"
-	}
+	// Only apply rules and default to records that doesn't point to a upstream record
```

Comment:
Good catch

Reply:
Probably could be refactored into a defaulting function, but not sure if that makes it simpler.